backend/src/appstate.rs

```diff
@@ -1,21 +1,52 @@
 use actix::Addr;
 
+use serde::Serialize;
 use std::sync::Mutex;
 
+use crate::api;
+use crate::communicator;
 use crate::game;
 use crate::websocket::Websocket;
 
 #[derive(Debug)]
 pub struct AppState {
     pub clients: Mutex<Vec<Addr<Websocket>>>,
     pub game: Mutex<game::Game>,
+    pub current_player: Mutex<u32>,
 }
 
 impl AppState {
     pub fn new() -> AppState {
         AppState {
             clients: Mutex::new(vec![]),
             game: Mutex::new(game::Game::new(0, 0)),
+            current_player: Mutex::new(0),
         }
     }
+
+    pub fn next_turn(&self) {
+        self.change_player();
+        self.send_turn();
+    }
+
+    pub fn broadcast<T: Serialize>(&self, msg: &T) {
+        let communicator = communicator::Communicator::new(self.clients.lock().unwrap().clone());
```

Comment:
I think we should add communicator to the app state.

Reply:
Yep, but it means that communicator holds pointer to vector of clients because they will update on appstate while new players (or second player) joining game